### ISSUE TITLE
Twitter not working even with updated service

### ISSUE THREAD

**Issue by zocheyado** [nraboy/ng-cordova-oauth#132]
So I was reading #45 to fix an error that I have when using twitter oauth Error 32 (Could not authenticate you). #45 talks about how to fix it by replacing the twitter code with 
```
$http({
        method: "post",
        url: "https://api.twitter.com/oauth/request_token",
         headers: {
                  "Authorization": signatureObj.authorization_header,
                  "Content-Type": "application/x-www-form-urlencoded"
         },
        data: "oauth_callback=" + encodeURIComponent("http://localhost/callback")
})
```
However the issue still persists even after this, I am using it on an actual android device not serve or live. I am also using code directly from documentation in the controller 
```
    $scope.twitterLogin = function() {
        $cordovaOauth.twitter('xxxxxxxx', 'xxxxxxxxx').then(function(result) {
            // results
            console.log("Success");
            console.log(result);
        }, function(error) {
            // error
            console.log("Failure");
            console.log(error);
        });
    }
```

**Comment by nraboy** [comment]
I'll try it again this weekend to see if something changed, but I get asked similar questions all the time and it always turns out to be a configuration issue rather than an issue with the library.

You might try this in the temporary:

https://github.com/nraboy/ng-cordova-oauth-example

@saimon24 also has written several tutorials and extensions for Twitter.  Including him in the conversation in case he has any opinions.

Best,

**Comment by jmtt89** [comment]
Hi, this change again 

```javascript
$http({
        method: "post",
        url: "https://api.twitter.com/oauth/request_token",
         headers: {
                  "Authorization": signatureObj.authorization_header,
                  "Content-Type": "application/x-www-form-urlencoded"
         },
        data: "oauth_callback=" + encodeURIComponent("http://localhost/callback")
})
```

Use without **data** 

```javascript
    var oauthObject = {
       oauth_consumer_key: clientId,
        oauth_nonce: $cordovaOauthUtility.createNonce(10),
        oauth_signature_method: "HMAC-SHA1",
        oauth_timestamp: Math.round((new Date()).getTime() / 1000.0),
        oauth_version: "1.0"
     };

     var signatureObj = $cordovaOauthUtility.createSignature("POST", "https://api.twitter.com/oauth/request_token", oauthObject,  { oauth_callback: redirect_uri }, clientSecret);

      $http({
          method: "post",
          url: "https://api.twitter.com/oauth/request_token",
          headers: {
               "Authorization": signatureObj.authorization_header,
               "Content-Type": "application/x-www-form-urlencoded"
          }
      })

```

and Twitter work fine

According the api doc:

OAuth parameters are not always sent the same way as other parameters. **If you’re using HTTP-header based OAuth, you shouldn’t include oauth_* parameters in the POST body or querystring.**

https://dev.twitter.com/oauth/reference/post/oauth/request_token

**Comment by nraboy** [comment]
I just tried it myself:

https://github.com/nraboy/ng-cordova-oauth-example

I have no problems, thus it being a configuration on your end with the Twitter dashboard.

Best,

**Comment by zocheyado** [comment]
So I followed your advice and tried doing nothing but ng-cordova-oauth-example and it still fails for me. So as you suggested you assume its something with my twitter dashboard app setup, any idea what needs to be done?  I am creating a brand new app, default everything, adding name and description and I still get the error. I have also tried adding http://localhost/callback as a callback and twitter does not let me.
